src/libs/actions/PersonalDetails.js

```diff
@@ -302,12 +302,25 @@ function fetchLocalCurrency() {
  */
 function setAvatar(file) {
     setPersonalDetails({avatarUploading: true});
-    API.User_UploadAvatar({file}).then((response) => {
-        // Once we get the s3url back, update the personal details for the user with the new avatar URL
-        if (response.jsonCode === 200) {
-            setPersonalDetails({avatar: response.s3url, avatarUploading: false}, true);
-        }
-    });
+    API.User_UploadAvatar({file})
+        .then((response) => {
+            // Once we get the s3url back, update the personal details for the user with the new avatar URL
+            if (response.jsonCode === 200) {
+                setPersonalDetails({avatar: response.s3url, avatarUploading: false}, true);
+            } else {
+                const error = new Error();
+                error.jsonCode = response.jsonCode;
+                throw error;
+            }
+        })
```

Comment:
Let's update this to use 
`if (response.jsonCode !== 200) {
	throw new Error(response.jsonCode);
}`

and then leave the setPersonalDetails({avatar: response.s3url, avatarUploading: false}, true); outside the if

Reply:
So after update .then() block will look as under:

```javascript
.then((response) => {
    // Once we get the s3url back, update the personal details for the user with the new avatar URL
    if (response.jsonCode !== 200) { throw new Error(response.jsonCode); }
    setPersonalDetails({avatar: response.s3url, avatarUploading: false}, true);
})
````

But here I tested and got problem with `throw new Error(response.jsonCode);`  It is not working as expected.

In case invalid file we are receiving response.jsonCode as 502, So this single line throw statement will create error object as string i.e. "Error: 502"  So .catch() block is receiving error object  as "Error: 502"   and due to that within .catch() block if () condition never match and always go to else part that shows profilePage.avatarUploadFailureMessage instead of showing invalidFileMessage 
```javascript
.catch((error) => {
    setPersonalDetails({avatarUploading: false});
    if (error.jsonCode === 405 || error.jsonCode === 502) {
        Growl.show(translateLocal('profilePage.invalidFileMessage'), CONST.GROWL.ERROR, 3000);
    } else {
        Growl.show(translateLocal('profilePage.avatarUploadFailureMessage'), CONST.GROWL.ERROR, 3000);
    }
});
````

**So I will suggest .then() block as under, so this will pass error as json object i.e. { "jsonCode": 502 } to catch() block. So within error block if() condition working as expected. i.e. When invalid file uploaded it shows invalid file message, and other jsonCode it shows upload failure message etc.**   I tested and this is working.
```javascript
.then((response) => {
    // Once we get the s3url back, update the personal details for the user with the new avatar URL
    if (response.jsonCode !== 200) { 
        const error = new Error();
         error.jsonCode = response.jsonCode;
        throw error;
    }
    setPersonalDetails({avatar: response.s3url, avatarUploading: false}, true);
})
````
Please let me know If this is ok so I can update and submit new commit.

Reply:
Ah yeah my bad, json code doesn't go where the error message does.  Yeah that updated solution is fine

Reply:
Though, there is an extra space in front of error.jsonCode that should be removed but that's probably just a formatting thing

Reply:
ok refactored code and pushed new commit. Also taken care for extra space in front of error.jsonCode. Thanks.